bothan-lib/src/telemetry/metrics.rs

```diff
@@ -0,0 +1,61 @@
+use opentelemetry::{global, KeyValue};
+
+pub struct Metrics{}
+
+impl Metrics {
+    pub fn increment_get_prices_total_requests() {
+        global::meter("server")
+        .u64_counter("get_prices_total_requests")
+        .with_description("total number of requests sent to fetch asset prices")
+        .build().add(1, &[]);
+    }
+
+    pub fn update_get_prices_responses(start_time: i64, success: bool) {
+        let elapsed_time = (chrono::Utc::now().timestamp_millis() - start_time) as u64; 
+        let meter = global::meter("server");
+        
+        meter.u64_counter("get_prices_total_responses")
+        .with_description("total number of responses received for asset price requests")
+        .build().add(1, &[KeyValue::new("success", success)]);
+
+        meter.u64_histogram("get_prices_response_time")
+        .with_description("time taken to fetch asset prices")
+        .with_unit("milliseconds")
+        .build().record(elapsed_time, &[KeyValue::new("success", success)]);
+    }
+
+    pub fn increment_rest_get_asset_info_total(source: &'static str, status: &'static str) {
```

Comment:
status seems to only be success and failure, please use an enum or bool instead of having the user pass in the explicit values